server/src/core/user/user-service.ts

```diff
@@ -61,6 +69,9 @@ export class Users implements UserService {
       _id: generateId(),
       password: await hashPassword(args.password),
       phone: args.phone,
+      addedBy: args.addedBy,
+      donors: args.role === 'donor' ? [] : [args.addedBy],
+      roles: [args.role],
```

Comment:
Let's remove `role` and `addedBy` from args. This method is only used for signing up, and you can only sign up as a donor so `donors` should be set to an empty array and `user.roles` hardcoded to `['donor']`